### ISSUE TITLE
Nano: Nano: add dataloader check for pytorch quantization to comply with INC calibration format

### ISSUE THREAD

**Issue by zhentaocc** [intel-analytics/BigDL#4163]
### Problem
I wrote a unit test to reproduce the error:
```python
from torch import nn
from torch.utils.data import TensorDataset, DataLoader

class MultipleInputs(nn.Module):
    def __init__(self):
        super().__init__()
        self.linear1 = nn.Linear(3, 1)
        self.linear2 = nn.Linear(4, 1)

    def forward(self, x1, x2):
        y1 = self.linear1(x1)
        y2 = self.linear2(x2)
        return y1.squeeze()

model = MultipleInputs()
x1 = torch.ones((10, 3))
x2 = torch.ones((10, 4))
y1 = torch.ones((10))
model(x1, x2)
dataset = TensorDataset(x1, x2, y1)
dataloader = DataLoader(dataset, batch_size=5)

trainer = Trainer()
trainer.quantize(model, dataloader)
```
With more than two tensors in a batch, e.g. `x1, x2, y1`,  INC requires to wrap them as `(x1, x2), y1`.  

### solution
A dataloader should be restricted to return no more than two items. It can be one of them:
- `(tensor_1, ... ,tensor_n), tensor_n1`
- `(tensor_1, ... ,tensor_n), (tensor_n1, ...,  tensor_n2)`
- `tensor_1, (tensor_2, ..., tensor_n)`
- `tensor_1, tensor_2`
- `tensor_1, tensor_2, ...` (no labels attached)

### What to do

- [x] Run sanity check on dataloader and raise error to instruct user to modify their dataloader
- [ ] (Optional) Provide a function to re-collate data from dataloader with index
- [x] Make sure everything remains fine, `trainer.fit(..)` `trainer.quantize(..)`

**Comment by zhentaocc** [comment]
Possible help function:
```python
        dataloader = DataLoader(dataset, batch_size=5)

        def input_label_split(collate_fn, index):
            def decorator(batch):
                batch = collate_fn(batch)
                x, y = batch[:index], batch[index:]
                if len(x) == 1:
                    x = x[0]
                if len(y) == 1:
                    y = y[0]
                return x, y
            return decorator

        dataloader.collate_fn = input_label_split(dataloader.collate_fn, 2)
        trainer = Trainer()
        trainer.quantize(model, dataloader)
```
